### ISSUE TITLE
Split bill - When delete Split Bill report from attendee DM & go back - Error page showing

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#29497]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.83-5

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

## Action Performed:
1. Open New Expensify app
2. Navigate to 1:1 DM with any of the Attendees you split bill with
3. Delete the IOU report
4. Tap Back button

## Expected Result:
User should be landed on LHN or 1:1 DM

## Actual Result:
"Hmm... It's not here" error page appears

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary> 

 

https://github.com/Expensify/App/assets/78819774/453fd44b-1d2d-47f4-94f8-ad8521716797

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by twisterdotcom** [comment]
Just a note @lanitochka17 that if it's reproducable already, feel free to add `External` when you create it as well.

**Comment by imbngy** [comment]
# Proposal

### **Please re-state the problem that we are trying to solve in this issue.**
404like page loads after deleting the money request

### **What changes do you think we should make in order to solve the problem?**
This, but the if should not be made on the chat report ID which may still exist, but on the whole chatReport Object, like in the code below this one:
```javascript
if (shouldDeleteIOUReport) {
    /*in this if im checking the reportID, which may not be null, so it should check the whole chatReport Object 
to see that if it has been deleted and if it is now null/undefined it should  go back to home screen / chatlist */
    if (iouReport.chatReportID == null || iouReport.chatReportID == undefined) {
        Navigation.goBack(ROUTES.HOME);
        Navigation.goBack(ROUTES.HOME); 
        //it should go back to the home screen / ChatList screen without loading the deleted report
    }
    else{
        Navigation.goBack(ROUTES.HOME);
        Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(iouReport.chatReportID));
    }
}
```

So the 2nd if should be something like:
```javascript
    if(!chatReportExists(iouReport.chatReportID)){
          Navigation.goBack(ROUTES.HOME);
          Navigation.goBack(ROUTES.HOME); 
          //it should go back to the home screen / ChatList screen without loading the deleted report
    } ...

function chatReportExists( chatReportID ){ ... }
```

**Comment by twisterdotcom** [comment]
@parasharrajat intrigued for your take here please.

**Comment by twisterdotcom** [comment]
Bump here @parasharrajat

**Comment by twisterdotcom** [comment]
Still waiting on proposals then I suppose.

**Comment by twisterdotcom** [comment]
Bump here @parasharrajat.

**Comment by twisterdotcom** [comment]
Can we assign @tienifr?

**Comment by iwiznia** [comment]
hmmmmm this seems like a duplicate of https://github.com/Expensify/App/issues/26569 which will be fixed by https://github.com/Expensify/App/pull/26498
@sobitneupane @adamgrzybowski @JmillsExpensify @mountiny can you confirm that indeed this is the same issue (looks like it to me) and that the PR will fix this?

**Comment by s77rt** [comment]
Seems like a dupe

**Comment by sobitneupane** [comment]
@s77rt has taken over as the C+ of the issue. He has more information about the issue.

**Comment by s77rt** [comment]
@parasharrajat This is a dupe and https://github.com/Expensify/App/pull/26498 will fix it
